stock_batch_picking/wizard/batch_picking_creator.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+# © 2012-2016 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import _, api, fields, models
+from openerp.exceptions import UserError
+
+
+class StockBatchPickingCreator(models.TransientModel):
+    """Create a stock.batch.picking from stock.picking
+    """
+
+    _name = 'stock.batch.picking.creator'
+    _description = 'Batch Picking Creator'
+
+    name = fields.Char(
+        'Name', required=True,
+        default=lambda x: x.env['ir.sequence'].next_by_code(
+            'stock.batch.picking'
+        ),
+        help='Name of the batch picking'
+    )
+    date = fields.Date(
+        'Date', required=True, select=True, default=fields.Date.context_today,
+        help='Date on which the batch picking is to be processed'
+    )
+
+    picker_id = fields.Many2one(
+        'res.users', string='Picker',
+        default=lambda self: self._default_picker_id(),
+        help='The user to which the pickings are assigned'
+    )
+
+    notes = fields.Text('Notes', help='free form remarks')
+
+    def _default_picker_id(self):
+        """ Return default_picker_id from the main company warehouse
+        except if a warehouse_id is specified in context.
+        """
+        warehouse_id = self.env.context.get('warehouse_id')
+        if warehouse_id:
+            warehouse = self.env['stock.warehouse'].browse(warehouse_id)
+        else:
+            warehouse = self.env['stock.warehouse'].search([
+                ('company_id', '=', self.env.user.company_id.id)
+            ], limit=1)
```

Comment:
You can remove after lines adding [:1] here:

warehouse = self.env['stock.warehouse'].search([
        ('company_id', '=', self.env.user.company_id.id)
    ], limit=1)**[:1]**